### ISSUE TITLE
Nieuw agendasysteem

### ISSUE THREAD

**Issue by Jille** [karpenoktem/kninfra#117]
Zodat we het kunnen koppelen aan planning en subscriptions; we een andere text voor leden kunnen gebruiken dan externen.

Het moet ook gesynct blijven worden naar Google Calendar en we willen ook ons archief van oude activiteiten niet verliezen.

**Comment by aykevl93** [comment]
Ik zie twee opties:

  * Agenda-items aanmaken in Google Calendar die dan automatisch aan subscriptions, planning en het fotoboek gelinkt kunnen worden.
  * Agenda-items in een eigen agenda aanmaken die we dan syncen naar Google Calendar

Allebei zijn te doen (zelfs tegelijkertijd), maar het laatste zou potentieel best gecompliceerd kunnen worden. Het eenvoudigste lijkt me om alle agenda-items lokaal te cachen (dat kan efficiënt met Calendar API v3, zie https://developers.google.com/google-apps/calendar/v3/sync). Dan kun je het (Calendar API) event ID opnemen in de subscription en daar leuke dingen mee doen.

**Comment by bwesterb** [comment]
Ik heb de voorkeur om het invoeren bij Google calendar te laten: een UI is namelijk lastiger te onderhouden dan een synchronisatiescriptje de andere kant op.  Een voordeel van lokale agenda items, is dat we niet meer afhankelijk zijn van Google, wat fijn is.  Maar goed, allemaal leuk en aardig, maar wat willen we aan functionaliteit toevoegen die we nu nog niet hebben?  Jille noemde het hebben van aparte tekst voor leden en externen, maar zou dat nu ook gebruikt worden?  Etc.

**Comment by aykevl93** [comment]
Ja, invoeren lijkt me ook makkelijker als we het bij Google Calendar laten. Als we het helemaal fancy willen doen, dan kunnen we ook push-notifications gebruiken, maar dat is puur omdat het leuk is en voegt niet zo heel veel toe (behalve gemak voor de secretaris).

Eventuele leuke functionaliteit die we in kunnen bouwen:

* Linkjes tussen agenda, subscriptions, planning en fotoboek.
* Kies bij het maken van een activiteit in het fotoboek uit een lijst in plaats van dat je alles handmatig moet invullen.
* We kunnen dezelfde tekst voor het event als voor de activiteit gebruiken. Ik weet niet of we dat willen en of het handig is.

Het zou handig kunnen zijn een overkoepelend systeem te bouwen voor activiteiten, al dan niet afhankelijk van de agenda. Ik zou alleen niet weten hoe dat er dan uit zou zien.
